Customizer: Fix input field widths and alignment for date-time controls

Infinite-Null opened the pull request:
Trac ticket: [#51249](https://core.trac.wordpress.org/ticket/51249)

## What
- Increased the width of date-time input fields from 46px to 52px
- Added vertical alignment to select fields in day and time controls

## Why
- The previous width of 46px was causing text to be truncated
- Select fields were misaligned with neighboring elements, creating visual inconsistency

## How
- Updated width property in date-time input styles to accommodate longer text
- Added vertical align: top to select elements within day and time field groups

## Testing Instructions
1. Go to Customizer
2. Navigate to any panel with date-time controls
3. Verify date/time input fields show complete numbers without truncation
4. Check that select fields are properly aligned with neighboring inputs

## Screenshots
#### Before patch:
![image](https://github.com/user-attachments/assets/4379c751-49a0-4cfb-b43a-af4af3ed8a1e)

#### After Patch:
![image](https://github.com/user-attachments/assets/82608bce-8305-46ab-98b8-2607d8ee016f)

Infinite-Null committed:
Customizer: Fix input field widths and alignment for date-time controls

Infinite-Null committed:
Customizer: Increase year input width in date-time fields

mukeshpanchal27 commented on src/wp-admin/css/customize-controls.css:
```diff
@@ -362,11 +362,16 @@ body.trashing #publish-settings {
+
+.wp-core-ui .date-time-fields .day-fields.clear select,
+.wp-core-ui .date-time-fields .time-fields.clear select {
+	vertical-align: top;
```
Nice

mukeshpanchal27 commented:
For input type number width it not look good normal state but when hover over the changes make sense.

<img width="309" alt="Screenshot 2025-01-23 at 4 42 19 PM" src="https://github.com/user-attachments/assets/9f1736fe-24b5-4b8c-b7c6-fa8a6da6a735" />

sabernhardt commented on src/wp-admin/css/customize-controls.css:
```diff
@@ -362,11 +362,16 @@ body.trashing #publish-settings {
+}
+
+.wp-core-ui .date-time-fields .day-fields.clear select,
+.wp-core-ui .date-time-fields .time-fields.clear select {
```
```suggestion
.customize-control-date_time select {
```

This only needs a class and an element to override `.wp-core-ui select { vertical-align: middle; }` in `forms.css`. The `.customize-control-date_time` class would cover both the date and the time.

sabernhardt commented on src/wp-admin/css/customize-controls.css:
```diff
@@ -362,11 +362,16 @@ body.trashing #publish-settings {
 .date-time-fields .date-input.hour,
 .date-time-fields .date-input.minute {
-	width: 46px;
+	width: 52px;
```
Instead of adding another 5 or 6 pixels to each input's width, you could add the `tiny-text` class on each of [these number fields](https://core.trac.wordpress.org/browser/trunk/src/wp-includes/customize/class-wp-customize-date-time-control.php?rev=55276&marks=157,167,185,193#L151). The `forms.css` stylesheet already [removes the right padding](https://core.trac.wordpress.org/browser/trunk/src/wp-admin/css/forms.css?rev=59046&marks=482,491#L480) to make room for the increment/decrement buttons. 

(`small-text` might seem more appropriate for the Year field, but it still has [padding on small screens](https://core.trac.wordpress.org/browser/trunk/src/wp-admin/css/forms.css?rev=59046&marks=1674-1685#L1684) and the Customizer's width works with `tiny-text`)

mukeshpanchal27 replied:
I like the idea to add the class and apply CSS their.

Infinite-Null committed:
Customize: Improve date/time input field styling

- Used WordPress core's tiny-text class for date/time number inputs instead of custom widths
- Provided better handling of increment/decrement buttons through core styles
- Ensured consistent styling across different screen sizes
- Simplified vertical alignment selector for date/time selects by targeting customize-control-date_time class directly

Infinite-Null commented:
Hi @sabernhardt and @mukeshpanchal27,
I have made the necessary changes to this PR. Please have a look at it at your convenience.
Thank you!

sabernhardt reviewed (approved):
Thanks!

Firefox shows the number field buttons without interaction, so I prefer that for the before and after screenshots at various widths.

![English screenshots](https://github.com/user-attachments/assets/09b4025e-b683-4031-a75e-c871b20d3e0a)
![French screenshots, with 24-hour time](https://github.com/user-attachments/assets/d1e13af0-fbf5-4ef4-8dc4-21e0a2f82531)
![Arabic screenshots](https://github.com/user-attachments/assets/cfe548f2-fd17-40ed-8d66-10de839488c5)

Chrome also shows the full digits with the patch, but the fields have an empty space until a user either hovers the cursor over the field or moves keyboard focus to it.

![Chrome, before and after, with empty space](https://github.com/user-attachments/assets/4c128327-5970-4f92-ac4d-8be36889a2b0)

I forced the focus state for all the fields to make all Chrome's buttons appear, with the patch applied.

![Chrome focus styles with patch applied](https://github.com/user-attachments/assets/884cffca-8784-4f4c-a6c3-059abba4f814)

audrasjb committed:
Random change just to restart GitHub Actions

audrasjb committed:
Random change just to restart GitHub Actions

audrasjb commented:
I restarted the unit tests since a lot of them were failing.

Infinite-Null commented:
> I restarted the unit tests since a lot of them were failing.

Hi @audrasjb,
It seems all the tests are passing now.

audrasjb commented:
committed in https://core.trac.wordpress.org/changeset/59942
